star-ros2/src/star_perception/star_perception/hailo_runner.py

```
@@ -0,0 +1,390 @@
+        )
+        output_vstreams_params = hp.OutputVStreamParams.make(
+            network_group, format_type=hp.FormatType.FLOAT32,
+        )
+        input_info = hef.get_input_vstream_infos()[0]
+
+        def backend(canvas: np.ndarray) -> list[tuple[int, float, float,
+                                                      float, float, float]]:
+            with network_group.activate(ng_params):
+                with hp.InferVStreams(network_group,
+                                      input_vstreams_params,
+                                      output_vstreams_params) as pipeline:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)